### ISSUE TITLE
BadFunctionCallException: metadata is not a valid attribute

### ISSUE THREAD

**Issue by AndyXheli** [nextcloud/recognize#806]
### Which version of recognize are you using?

3.7.0

### Enabled Modes

Face recognition

### TensorFlow mode

Normal mode

### Which Nextcloud version do you have installed?

26.0.1

### Which Operating system do you have installed?

Ubuntu 22.0.4

### Which Docker container are you using to run Nextcloud? (if applicable)

N/A

### How much RAM does your server have?

24

### What processor Architecture does your CPU have?

X64

### Describe the Bug

Stuck on Face recognition: 7571 Queued files, Last classification: 11 hours ago nothing happens. When i click on photos then people i get blank 

![image](https://user-images.githubusercontent.com/59488153/233399883-e54b7c6b-ac07-41c9-baf4-a3c495bfdbd1.png)

### Expected Behavior

Should display data

### To Reproduce

Not sure

### Debug log

```
{"reqId":"LnV9WvzVRx8G8ve2jS7A","level":3,"time":"2023-04-20T09:14:32-05:00","remoteAddr":"8.8.8.8","user":"admin","app":"webdav","method":"PROPFIND","url":"/remote.php/dav/recognize/admin/faces/admin%20X/","message":"metadata is not a valid attribute","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36","version":"26.0.1.1","exception":{"Exception":"BadFunctionCallException","Message":"metadata is not a valid attribute","Code":0,"Trace":[{"file":"/var/www/nextcloud/lib/public/AppFramework/Db/Entity.php","line":169,"function":"getter","class":"OCP\\AppFramework\\Db\\Entity","type":"->"},{"file":"/var/www/nextcloud/apps/recognize/lib/Dav/Faces/FacePhoto.php","line":148,"function":"__call","class":"OCP\\AppFramework\\Db\\Entity","type":"->"},{"file":"/var/www/nextcloud/apps/recognize/lib/Dav/Faces/PropFindPlugin.php","line":57,"function":"getMetadata","class":"OCA\\Recognize\\Dav\\Faces\\FacePhoto","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/PropFind.php","line":95,"function":"OCA\\Recognize\\Dav\\Faces\\{closure}","class":"OCA\\Recognize\\Dav\\Faces\\PropFindPlugin","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/nextcloud/apps/recognize/lib/Dav/Faces/PropFindPlugin.php","line":57,"function":"handle","class":"Sabre\\DAV\\PropFind","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"propFind","class":"OCA\\Recognize\\Dav\\Faces\\PropFindPlugin","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":1052,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":984,"function":"getPropertiesByNode","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":1662,"function":"getPropertiesIteratorForPath","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":1647,"function":"writeMultiStatus","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php","line":346,"function":"generateMultiStatus","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"httpPropFind","class":"Sabre\\DAV\\CorePlugin","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":472,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":321,"function":"start","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/Server.php","line":366,"function":"exec","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/apps/dav/appinfo/v2/remote.php","line":35,"function":"exec","class":"OCA\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/remote.php","line":172,"args":["/var/www/nextcloud/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/var/www/nextcloud/lib/public/AppFramework/Db/Entity.php","Line":152,"message":"metadata is not a valid attribute","exception":[],"CustomMessage":"metadata is not a valid attribute"},"id":"64414d1f0605f"}

```

**Comment by AndyXheli** [comment]
Thank you for confirming! Should we disable the app? or its okay to leave it on ?

**Comment by crocobc** [comment]
Confirmed also (same versions)

**Comment by Andreaux** [comment]
Has anyone gotten closer to a cause or solution?

**Comment by marcelklehr** [comment]
The problem is that recognize depends on an OC API which, as opposed to OCP, is not public and thus not stable. In Nextcloud 25 and nextcloud 26.0.0 The getMetadata() Method was available, but since Nextcloud v26.0.1 it was renamed.

I suppose changing the code in recognize and dropping support for 25 would be the best past forward. We also don't need to drop support for 25 entirely, but have to confine it to a stable25 branch. I'll look into this tomorrow!

**Comment by Andreaux** [comment]
If there's anything I can do to help, I'd be happy to. Thanks for your help!
